api/sites.ts

```diff
@@ -89,6 +89,30 @@ export const siteRoutes = (cfg: APIConfig, store: StoreI) => async (server: Fast
     return await reply.send(await store.sites.get(id))
   })
 
+  server.get<{
+    Params: {
+      id: string
+    }
+    Reply: Static<typeof SiteStats>
+  }>('/sites/:id/stats', {
+    schema: {
+      params: Type.Object({
+        id: Type.String()
+      }),
+      response: {
+        200: SiteStats
+      },
+      description: 'Returns the stats for a site',
+      tags: ['site'],
+      security: [{ jwt: [] }]
+    },
+    preHandler: server.auth([server.verifyPublisher, server.verifyAdmin])
+  }, async (request, reply) => {
+    const { id } = request.params
+    const stats = await store.sites.stats(id)
+    return await reply.send(stats)
```

Comment:
We can add more logs

```suggestion
// Logging the request initiation  
request.log.info(`Fetching stats for site ID: ${id}`)
  try {
    const stats = await store.sites.stats(id)
    // Logging the successful retrieval of stats
    request.log.info(`Successfully retrieved stats for site ID: ${id}`, { stats })
    return await reply.send(stats)
  } catch (error) {
    // Logging any errors that occur during the stats retrieval
    request.log.error(`Error fetching stats for site ID: ${id}`, { error })
    return reply.status(500).send({ error: 'Failed to retrieve site stats' })
  }
```